### PR TITLE
fix: lifecycle-expiration validation bug

### PR DESCRIPTION
**Pull request by KevinSmile**
## Description
Currently, when you set a lifecycle  to a bucket:
1. you can set Expiration with both Days and Date, only Date works.
image

2. when you set Days=0, the error message is confusing.
image

It seems that:
1. unit-tests covered the above two cases in `expiration_test.go.go`, but not in`lifecycle_test.go`.
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/pkg/bucket/lifecycle/expiration_test.go#L26
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/pkg/bucket/lifecycle/lifecycle_test.go#L27

2. cases covered in xml-Unmarshal not working when you do http-body decode
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/pkg/bucket/lifecycle/expiration.go#L43-L45
https://github.com/minio/minio/blob/95411228db62ca90907987511c325874b4362b78/cmd/bucket-lifecycle-handlers.go#L69

## Motivation and Context
improve user experience

## How to test this PR?
set a lifecycle  to a bucket:
1. set Expiration with both Days and Date, and check the error message
2. set Expiration Days=0, and check the error message

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Documentation needed
- [ ] Unit tests needed

### CODE REVIEW

**Comment by KevinSmile** [comment]
BTW, currently the Days of an Expiration must be a positive integer, but the Date can be a past date (e.g. this day last year )?

**Comment by harshavardhana** [comment]
> BTW, currently the Days of an Expiration must be a positive integer, but the Date can be a past date (e.g. this day last year )?

Yes that is correct @KevinSmile

File: pkg/bucket/lifecycle/rule.go [hunk 1]
```diff
@@ -49,7 +49,6 @@ var (
 	errInvalidRuleID           = Errorf("ID length is limited to 255 characters")
 	errEmptyRuleStatus         = Errorf("Status should not be empty")
 	errInvalidRuleStatus       = Errorf("Status must be set to either Enabled or Disabled")
-	errMissingExpirationAction = Errorf("No expiration action found")
```
Comment (KevinSmile) [inline pkg/bucket/lifecycle/rule.go:52]: you never know whether the Expiration is missing or Days=Date=0

**Review by harshavardhana — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-gateway-s3.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-zoned.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-gateway-azure.sh | :x: [more...](#ccaaa9b0-cc80-434b-a00d-5c62505a3153) |
<a name="ccaaa9b0-cc80-434b-a00d-5c62505a3153"></a>
### 10327-0e73fe9/mint-gateway-azure.sh.log:
```
Running with
SERVER_ENDPOINT:      minio-dev4.minio.io:30869
ACCESS_KEY:           minioazure
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 745d50382175:/mint/log /tmp/mint-logs'

(1/15) Running aws-sdk-go tests ... done in 6 seconds
(2/15) Running aws-sdk-java tests ... done in 2 seconds
(3/15) Running aws-sdk-php tests ... done in 4 minutes and 57 seconds
(4/15) Running aws-sdk-ruby tests ... done in 16 seconds
(5/15) Running awscli tests ... FAILED in 1 minutes and 56 seconds
{
  "name": "awscli",
  "duration": 1485,
  "function": "make_bucket\n",
  "status": "FAIL",
  "error": "An error occurred (BucketAlreadyOwnedByYou) when calling the CreateBucket operation: Your previous request to create the named bucket succeeded and you already own it."
}
(5/15) Running healthcheck tests ... done in 0 seconds
(6/15) Running mc tests ... done in 4 minutes and 4 seconds
(7/15) Running minio-dotnet tests ... done in 2 minutes and 37 seconds
(8/15) Running minio-go tests ... FAILED in 5 minutes and 5 seconds
{
  "args": {},
  "duration": 1434,
  "error": "The specified multipart upload does not exist. The upload ID may be invalid, or the upload may have been aborted or completed.",
  "function": "CopyObjectPart(destination, source)",
  "message": "CompleteMultipartUpload call failed",
  "name": "minio-go: testUnencryptedToSSES3CopyObjectPart",
  "status": "FAIL"
}
(8/15) Running minio-java tests ... done in 6 minutes and 7 seconds
(9/15) Running minio-js tests ... done in 2 minutes and 37 seconds
(10/15) Running minio-py tests ... done in 10 minutes and 15 seconds
(11/15) Running s3cmd tests ... done in 1 minutes and 28 seconds
(12/15) Running s3select tests ... done in 51 seconds
(13/15) Running security tests ... done in 0 seconds

Executed 13 out of 15 tests successfully.
```
Deleting image on docker hub
Deleting image locally
Error: No such image: minio/minio:10327-0e73fe9

**Review by vadmeste — APPROVED** [review]
LGTM

**Pull request merged**